### PR TITLE
Migrate to tf v2 compatible

### PR DESCRIPTION
**Pull request by zyyang**
Migrate to tf v2 compatible.

---
Tested:
Original V1.* training result:
{
	"dataset": "polyvore",
	"learning_rate": 0.001,
	"weight_decay": 0.0,
	"epochs": 4000,
	"hidden": [350, 350, 350, 350],
	"dropout": 0.5,
	"degree": 1,
	"summaries_dir": "logs/",
	"support_dropout": 0.15,
	"write_summary": false,
	"batch_norm": true,
	"amz_data": "Men_bought_together",
	"best_val_score": 0.9276251196861267,
	"best_epoch": 3954,
	"best_epoch_train_score": 0.8384166955947876,
	"best_train_score": 0.840186357498169,
	"seed": 1572082204
}

V2 comp training result:
{
	"dataset": "polyvore",
	"learning_rate": 0.001,
	"weight_decay": 0.0,
	"epochs": 4000,
	"hidden": [350, 350, 350, 350],
	"dropout": 0.5,
	"degree": 1,
	"summaries_dir": "logs/",
	"support_dropout": 0.15,
	"write_summary": false,
	"batch_norm": true,
	"amz_data": "Men_bought_together",
	"best_val_score": 0.9242312908172607,
	"best_epoch": 3982,
	"best_epoch_train_score": 0.8369424939155579,
	"best_train_score": 0.8399352431297302,
	"seed": 1572506422
}

### CODE REVIEW

**Comment by gcucurull** [comment]
Thank you so much for this improvement, I am wondering whether to merge this into master or create a new branch for TF 2, this way people can still use the 1.X version if they haven't migrated to TF 2 yet.

What do you think?

**Comment by zyyang** [comment]
That's what I am planning to do.
I need, also, to make sure the test scripts still work, and make it running on real tf 2 api.
Let's close this pull request.
